### PR TITLE
Remove references to docker.

### PR DESCRIPTION
**Pull request by rhatdan**
Docker image -> container image
Docker containre -> container
Docker registry -> container image registry

### CODE REVIEW

File: day_two_guide/topics/pruning_images_and_containers.adoc [hunk 1]
```diff
@@ -204,12 +204,12 @@ $ sudo docker rmi *<my_manually_pulled_image:tag>*
...
 
 ----
-$ sudo docker rmi $(sudo docker images -f "dangling=true" -q)
+$ sudo docker rmi $(sudo container images -f "dangling=true" -q)
 ----
 
 WARNING: While not recommended, a more aggressive clean up can be performed
 by attempting to delete all the images.
-`docker` protects images it is using; this can be performed as `sudo docker rmi $(sudo docker images -q)`
+`docker` protects images it is using; this can be performed as `sudo docker rmi $(sudo container images -q)`
```
Comment (kalexand-rh) [inline day_two_guide/topics/pruning_images_and_containers.adoc]: need to revert this one.

File: dev_guide/application_lifecycle/promoting_applications.adoc [hunk 2]
```diff
@@ -28,7 +28,7 @@ those language specific runtimes.
 In the context of {product-title} and its combined foundation of Kubernetes and
 Docker, additional application artifacts include:
 
-* _Docker container images_ with their rich set of metadata and associated tooling.
+* _container images_ with their rich set of metadata and associated tooling.
```
Comment (kalexand-rh) [inline dev_guide/application_lifecycle/promoting_applications.adoc]: Caps here

File: rest_api/api/v1.Pod.adoc [hunk 3]
```diff
@@ -164,9 +164,9 @@ Populated by the system. Read-only. More info: http://kubernetes.io/docs/user-gu
...
 </div></details></details></details><div style="margin-left:13px;">  <span title="(boolean) AutomountServiceAccountToken indicates whether a service account token should be automatically mounted.">automountServiceAccountToken</span>:
 </div><details><summary>  <span title="(array) List of containers belonging to the pod. Containers cannot currently be added or removed. There must be at least one container in a Pod. Cannot be updated.">containers</span>:
-</summary><details><summary>  - <span title="(array) Arguments to the entrypoint. The docker image&#39;s CMD is used if this is not provided. Variable references $(VAR_NAME) are expanded using the container&#39;s environment. If a variable cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded, regardless of whether the variable exists or not. Cannot be updated. More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell">args</span>:
```
Comment (kalexand-rh) [inline rest_api/api/v1.Pod.adoc]: We don't manually change the API doc, so all these need to be reverted.

**Review by kalexand-rh — COMMENTED** [review]
@rhatdan, we don't manually update the API docs. Will you please revert changes to those topics?

I see two other changes.

File: day_two_guide/topics/pruning_images_and_containers.adoc (hunk 1 above)
Comment (rhatdan) [reply day_two_guide/topics/pruning_images_and_containers.adoc]: Fixed,

File: dev_guide/application_lifecycle/promoting_applications.adoc (hunk 2 above)
Comment (rhatdan) [reply dev_guide/application_lifecycle/promoting_applications.adoc]: Fixed

File: rest_api/api/v1.Pod.adoc (hunk 3 above)
Comment (rhatdan) [reply rest_api/api/v1.Pod.adoc]: Reverted

**Comment by kalexand-rh** [comment]
@rhatdan, thanks! This is almost good to merge - all the changes in the /rest_api directory need to be reverted first.

**Comment by kalexand-rh** [comment]
Thank you @rhatdan!

The build's clean, so please squash your commits so I can merge. :)

**Pull request merged**